### ISSUE TITLE
Semantics polish meta bug

### ISSUE THREAD

**Issue by abarth** [flutter/flutter#4792]
- [x] Crashes
  - [x] https://github.com/flutter/flutter/issues/2333
- [x] Modality
  - [x] https://github.com/flutter/flutter/issues/1705
  - [x] https://github.com/flutter/flutter/issues/3821
  - [x] https://github.com/flutter/flutter/issues/3819
- [ ] AX doesn't work if already enabled
- [x] Be able to turn off https://github.com/flutter/flutter/issues/3342
- [x] hasSemantics -> isSemanticBoundary
- [ ] Move TestSemanticsClient to flutter_test
- [x] SemanticsNode.addListener needs a "remove" companion
- [ ] Value for adjustable

**Comment by goderbauer** [comment]
@abarth @Hixie Do you remember what was meant by these points:

* AX doesn't work if already enabled
* Move TestSemanticsClient to flutter_test (TestSemanticsClient doesn't have any hits in the code base)
* Value for adjustable

**Comment by goderbauer** [comment]
I am closing this bug for now as I can't put it in the new accessibility project as is. Feel free to open new bugs for the unresolved issues and label them with accessibility.
